Copy ADR template from Tendermint

ValarDragon opened the pull request:
I believe we've been discussing adding these templates for awhile. I think its more amenable for longer proposals than github issues (as it allows threads of conversation), and was also desired in #1192
______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

alexanderbez commented on docs/architecture/adr-template.md:
```diff
@@ -0,0 +1,15 @@
+# ADR 000: Template for an ADR
```
Do you think the template I had [here](https://github.com/cosmos/cosmos-sdk/pull/1882/commits/6d75b69a5c937814b700a830045bdb710a5c8f67#diff-5be5997d8f94baad927ca518d17f9417) would be more helpful/descriptive?

ValarDragon replied:
Oh cool didn't see that! Yeah that template looks way better, I'll copy it here

alexanderbez replied:
✊

ValarDragon commented:
I modified @alexanderbez ADR Template. I modified it a fair amount though, from my understanding with the Tendermint ADR's. Feel free to comment if anyone would like any of the section descriptors to be changed!
